built_value_generator/lib/src/value_source_class.dart

```diff
@@ -933,11 +933,13 @@ abstract class ValueSourceClass
     result.writeln('  return other is $name${forBuilder ? 'Builder' : ''}');
     if (comparedFields.isNotEmpty) {
       result.writeln('&&');
-      result.writeln(comparedFields
-          .map((field) => field.isFunctionType
-              ? '${field.name} == _\$dynamicOther.${field.name}'
-              : '${field.name} == other.${field.name}')
-          .join('&&'));
+      result.writeln(comparedFields.map((field) {
+        var nameOrThisDotName =
+            field.name == 'other' ? 'other.${field.name}' : field.name;
```

Comment:
Should it be `'this.'`?

Reply:
I think it would be easier to read just like this, without string substitution: `field.name == 'other' ? 'this.other' : field.name`

Reply:
Oh man, that's embarrassing. Thanks. Done (both), and updated the test to test _not equal_ as well as _equal_.